tests/api/test_attributes.py

```diff
@@ -98,6 +98,35 @@ def test_attributes_query(user_api_client, product):
     assert len(attributes_data) == attributes.count()
 
 
+NOT_EXISTS_IDS_ATTRIBUTES_QUERY = """
+    query ($filter: AttributeFilterInput!) {
+        attributes(first: 5, filter: $filter) {
+            edges {
+                node {
+                    id
+                    name
+                }
+            }
+        }
+    }
+    """
+
+
+@pytest.mark.parametrize(
+    "attribute_filter", [{"ids": ["ygRqjpmXYqaTD9r=", "PBa4ZLBhnXHSz6v="]}],
+)
+def test_attributes_query_ids_not_exists(attribute_filter, user_api_client, category):
+    query = NOT_EXISTS_IDS_ATTRIBUTES_QUERY
+    variables = {"filter": attribute_filter}
+    response = user_api_client.post_graphql(query, variables)
+    content = get_graphql_content(response, ignore_errors=True)
+    message_error = "{'ids': [{'message': 'Invalid ID specified.', 'code': ''}]}"
+
+    assert content["errors"]
+    assert content["errors"][0]["message"] == message_error
```

Comment:
Are you sure about this? The message shouldn't be a string but a dict

Reply:
it's the data type that came from the response. if you want we can cast that string to a dictionary before comparing them in those tests.
<img width="646" alt="Screen Shot 2020-04-09 at 12 05 12 PM" src="https://user-images.githubusercontent.com/24206382/78915730-5fbba600-7a5a-11ea-9d87-47be4d7d262f.png">

Reply:
So the issue is that GraphQL we _cannot_ tell GraphQL or the client that they can retrieve a `errors` field, like done in mutations. So we only can use the current structure and typings, that are:
```yaml
errors:
  - message: str
    locations:
      - line: int
        column: int
```

So we should return a string as message. And not a python representation of a dictionary (that's totally wrong, we should definitely not do that; and it is an invalid parameter type for GraphQL, it is expecting a string, not a dict).

It is the only possibility we have.

Reply:
It's valid - it is a dict but wrapped in a string. If @gabmartinez is willing to keep improving that, we can do that but let's agree on what format we want to have in the `message` field. I'm also fine with the current solution as it's better what we had.

Reply:
@maarcingebala @NyanKiyoshi I think we can create another task to work on that because maybe we are going to need more time to review what we can do.

Reply:
In that case, if we do want a dict, we have to pass a valid JSON object, not a python representation of a Python dictionary, that's invalid JSON. On top, we need to pass it a string not as a dict to GraphQL, it is expecting a string.

Overall, I don't like the idea of passing a JSON object inside the error message, as it is a simple error telling the developer that they passed something invalid. We shouldn't add an expensive JSON serialization just to tell the developer they passed an invalid ID type.

Reply:
sorry, if I did wrong, but the `message_error` variable in those tests is not an argument that we are passing to GraphQL. We use it to compare the result content. Please let me know if that was you were worrying.

Reply:
When you raise `GraphQLError(instance.errors.get_json_data())`, it is expecting a string, not a dict.

Reply:
Gotcha, it is correct. So, your suggestion is to put all of them in one message string. Quick question, if more than one field is not valid, which format should we use to return the message, separate them by a comma?

Like this? `field_1: message_1, field_2: message_2`

Reply:
I don't understand the question. The signature is:
```
GraphQLError(message: str, nodes, ...)
```

Reply:
ok, thinking in the future if we have other fields to filter the result and more than one field is not valid in the input, how should we return the message error in one string. 

For example, the  `instance.errors` has a value like this `"{'ids': [ValidationError(['Invalid ID specified.'])]}, {'names': [ValidationError(['Invalid data type specified.'])]}"`, the response error message should be `ids: Invalid ID specified., names: Invalid data type specified.` in one string?

If it doesn't matter update it.

Reply:
For now, dump the dict to JSON and pass it into `message`

Reply:
gotcha, one it. :)